### PR TITLE
os/arch/arm/src/amebad fix for UART test case issue

### PR DESCRIPTION
**Pull request by prashantrar**
This PR contains the following changes:

1. Changes to fix certain UART test cases.

All commits have been labelled with respective changes.

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@prashantrar Thank you for raising the PR. But I don't know why you modify this. Could you fill the problem and reason in commit description?

**Comment by sunghan-chang** [comment]
And please don't mix the different kind of change. This commit includes the change on operation and the change of coding style. It makes us difficult to find what change is.

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 1]
```diff
@@ -260,16 +252,21 @@ static char g_uart2txbuffer[CONFIG_UART2_TXBUFSIZE];
...
-	.StickParity = RUART_STICK_PARITY_DISABLE,
-	.StopBit = CONFIG_UART0_2STOP,
-	.WordLen = CONFIG_UART0_BITS,
-	.FlowControl = FlowControlNone,
-	.irq = RTL8721D_UART0_IRQ,
+
+	.parity = CONFIG_UART0_PARITY,
+	.bits = CONFIG_UART0_BITS,
+#if (CONFIG_UART0_2STOP)
+	.stopbit = 2,
```
Comment (thapav) [inline os/arch/arm/src/amebad/amebad_serial.c:259]: Use #defines , avoid hardcoding

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 2]
```diff
@@ -423,36 +425,24 @@ static void rtl8721d_up_shutdown(struct uart_dev_s *dev)
...
-		/* Enable the interrupt (RX and TX interrupts are still disabled
-		 * in the UART
-		 */
-
-		up_enable_irq(priv->irq);
-	}
-#else
-	serial_irq_handler(sdrv[uart_index_get(priv->Tx)], rtl8721d_uart_irq, (uint32_t)dev);
-#endif
+	serial_irq_handler(sdrv[uart_index_get(priv->tx)], rtl8721d_uart_irq, (uint32_t) dev);
```
Comment (thapav) [inline os/arch/arm/src/amebad/amebad_serial.c:445]: Is the UART interrupt working with these changes ??

File: os/arch/arm/src/amebad/amebad_serial.c (hunk 1 above)
Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_serial.c:259]: Resolved in new commit

File: os/arch/arm/src/amebad/amebad_serial.c (hunk 2 above)
Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_serial.c:445]: HAve updated only UART test case changes in new commit

**Comment by prashantrar** [comment]
@sunghan-chang I have created a new commit with only functional changes

**Comment by prashantrar** [comment]
@AdamZhang0124 Can you comment on the functional changes in this PR

**Comment by AdamZhang0124** [comment]
This PR: 1) Edited parameters in ‘rtl8721d_up_dev_s’ and ‘rtl8721d_up_ioctl’ functions. 2) Removed unnecessary log.
It is part of the uart porting, without these changes, some of uart test cases(loop back tests) will fail.
To conduct uart itc and utc tests, it need to follow board specific settings and connection.

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 3]
```diff
@@ -167,36 +167,28 @@
  * UART2_DEV: KM0 log uart
  * UART3_DEV: KM0 luart
  */
+int txint_enable = 0;
```
Comment (sijupunnnoose) [inline os/arch/arm/src/amebad/amebad_serial.c:170]: Is it right to use a global variable for this? It should be part of the rtl8721d_up_dev_s??

Comment (AdamZhang0124) [reply os/arch/arm/src/amebad/amebad_serial.c:170]: Thanks for the feedback, let me check on this.

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 4]
```diff
@@ -167,36 +167,28 @@
...
-	uint32_t irq;                           /* IRQ associated with this UART */
-	PinName Tx;				/* TX UART pin number */
-	PinName Rx;				/* RX UART pin number */
+
+	uint8_t parity;				/* 0=none, 1=odd, 2=even */
+	uint8_t bits;				/* Number of bits (7 or 8) */
+	uint8_t stopbit;			/* Number of StopBit (1 or 2) */
+	uint32_t baud;				/* Configured baud rate */
+	uint32_t irq;				/* IRQ associated with this UART */
+	PinName tx;					/* TX UART pin number */
```
Comment (deepaksrma) [inline os/arch/arm/src/amebad/amebad_serial.c:181]: Please maintain the same tab space

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 5]
```diff
@@ -167,36 +167,28 @@
...
+	PinName rx;					/* RX UART pin number */
```
Comment (deepaksrma) [inline os/arch/arm/src/amebad/amebad_serial.c:182]: same as above

File: os/arch/arm/src/amebad/amebad_serial.c (hunk 4 above)
Comment (AdamZhang0124) [reply os/arch/arm/src/amebad/amebad_serial.c:181]: sure, I am fixing on that.

**Comment by AdamZhang0124** [comment]
Hi @sijupunnnoose and @deepaksrma ,
I have updated the code, please help to review. Thanks!

**Pull request merged**